File: chapter01/StringRotation.ts
function StringRotation(s1: string, s2: string): boolean {
    if (s1.length !== s2.length) return false;

    return isSubString(s1, s2);
}

function isSubString(s1: string, s2: string): boolean {
    return (s2 + s2).includes(s1);
}

console.log(StringRotation("waterbottle", "erbottlewat"));
console.log(StringRotation("itakistakis", "erbtotlewat"));

File: chapter01/stringRotation/StringRotation.java
public class StringRotation {
    public static void main(String[] args) {
        String s1 = "erbottlewat";
        String s2 = "waterbottle";
        System.out.println(isSubstring(s1, s2));
    }

    public static boolean isSubstring(String s1, String s2) {
        if (s1.length() != s2.length()) {
            return false;
        }

        String s1s1 = s1 + s1;

        return s1s1.contains(s2);
    }
}
